rosbag2_test_common/include/rosbag2_test_common/client_manager.hpp

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 Sony Group Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TEST_COMMON__CLIENT_MANAGER_HPP_
+#define ROSBAG2_TEST_COMMON__CLIENT_MANAGER_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rcl/service_introspection.h"
+
+#include "rclcpp/rclcpp.hpp"  // rclcpp must be included before the Windows specific includes.
+
+
+namespace rosbag2_test_common
+{
+template<typename ServiceT>
+class ClientManager : public rclcpp::Node
+{
+public:
+  explicit ClientManager(
+    std::string service_name,
+    size_t client_number = 1,
```

Comment:
Comments from MichaelOrlov

The name for the parameter client_number_ is misleading. One could think that this is some sequence number for the client or an index of the client in some inner array.
The more appropriate name would be number_of_clients_. @Barry-Xu-2018 Please rename it.

Reply:
Updated